flow/embedded_views.cc

```diff
@@ -6,7 +6,46 @@
 
 namespace flutter {
 
+ExternalViewEmbedder::ExternalViewEmbedder()
+    : transformStack(std::make_shared<EmbeddedViewMutatorStack>()){};
+
 bool ExternalViewEmbedder::SubmitFrame(GrContext* context) {
   return false;
 };
+
+void EmbeddedViewMutatorStack::pushClipRect(const SkRect& rect) {
+  EmbeddedViewMutator element = EmbeddedViewMutator();
+  element.setType(clip_rect);
+  element.setRect(rect);
+  vector_.push_back(element);
```

Comment:
nit: we can save some copies by making it a vector of unique pointers

Reply:
Adding a TODO and will work on it in future.